utilities/include/utilities/log.h

```diff
@@ -124,22 +145,31 @@ class Logger {
    * @param message Log message
    */
   void log(LogLevel level, const char *format, va_list args) noexcept {
-    if (level < logLevel_) {
+    if ((level < log_level_) || (level == LogLevel::DISABLED)) {
       return;
     }
 
-    std::lock_guard<std::mutex> guard(mutex_log_);
-    char logMessage[1024]{};
-    vsnprintf(logMessage, sizeof(logMessage), format, args);
+    char log_msg[1024]{};
+    vsnprintf(log_msg, sizeof(log_msg), format, args);
```

Comment:
Check output of `vsnprintf`

Reply:
ignore this because exceptions are not handled. This should be fix in c++ version